### PR TITLE
fix: [CI-4002]: Pass context to the APIs in the TI service HTTP Client

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32556)

### CODE REVIEW

**Comment by rutvijmehta-harness** [comment]
trigger messagecheck

File: product/ci/ti-service/client/http.go [hunk 1]
```diff
@@ -90,23 +90,23 @@ func (c *HTTPClient) DownloadLink(ctx context.Context, language, os, arch, frame
...
 	return resp, err
 }
 
 // UploadCg uploads avro encoded callgraph to server
-func (c *HTTPClient) UploadCg(org, project, pipeline, build, stage, step, repo, sha, source, target string, timeMs int64, cg []byte) error {
+func (c *HTTPClient) UploadCg(ctx context.Context, org, project, pipeline, build, stage, step, repo, sha, source, target string, timeMs int64, cg []byte) error {
 	path := fmt.Sprintf(cgEndpoint, c.AccountID, org, project, pipeline, build, stage, step, repo, sha, source, target, timeMs)
-	ctx := context.WithValue(context.Background(), "reqId", sha)
+	ctx = context.WithValue(ctx, "reqId", sha)
 	backoff := createBackoff(45 * 60 * time.Second)
```
Comment (hemanthmantri) [inline product/ci/ti-service/client/http.go:110]: Let's create a ticket to use flags/consts for magic numbers in the TI code.

**Review by hemanthmantri — COMMENTED** [review]
You need to add `go` label to the PR to trigger to tests. It is not part of harness pipeline tests.

File: product/ci/ti-service/client/http.go (hunk 1 above)
Comment (rutvijmehta-harness) [reply product/ci/ti-service/client/http.go:110]: Done: https://harness.atlassian.net/browse/CI-4324

**Review by hemanthmantri — APPROVED** [review]
LGTM

**Review by hemanthmantri — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32556)* status: 1. unresolved discussions.

**Review by vistaarjuneja — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32556#-:-N0lfns701uikY3-I2rJ:bnfp4nl)* status: 1. unresolved discussions.

**Review by jamie-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32556)* status: Some of the checks are too old. 1. unresolved discussions.

**Review by hemanthmantri — APPROVED** [review]

**Review by hemanthmantri — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32556)* status: Some of the checks are too old. 1. unresolved discussions.

**Review by hemanthmantri — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32556)* status: Some of the checks are too old.

**Review by hemanthmantri — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32556#-:-N1EuwTz6TOL8eieZgOB:bnfp4nl)* status: Some of the checks are too old.

**Pull request merged**